src/game/RandomizationOfShipsPositions.js

```diff
@@ -0,0 +1,152 @@
+import { SHIPS_LIST } from "./constants";
+
+const GAME_BOARD = document.querySelectorAll(".board__fields");
+
+let gameBoardArray = [];
+let fieldsNamesArr = [];
+let direction;
+let position;
+const GAME_BOARD_WIDTH = 10;
+
+const createFieldsName = () => {
+  for (let i = 0; i < 10; i++) {
+    for (let j = 0; j < 10; j++) {
+      fieldsNamesArr.push(`${String.fromCharCode(i + 65)}${j + 1}`);
+    }
+  }
+};
+createFieldsName();
+
+const createGameBoardArray = () => {
+  gameBoardArray = [];
+  for (let i = 0; i < GAME_BOARD_WIDTH; i++) {
+    gameBoardArray.push([]);
+    for (let j = 0; j < GAME_BOARD_WIDTH; j++) {
+      gameBoardArray[i].push({
+        isEmpty: true,
+        shipName: "",
+      });
+    }
+  }
+};
+
+const directionRoll = () => {
+  Math.floor(Math.random() * 2) === 0
+    ? (direction = "vertical")
+    : (direction = "horizontal");
+};
+
+const makePosition = (shipIndex) => {
+  position = [];
+  for (let i = 0; i < 2; i++) {
+    position.push(Math.floor(Math.random() * GAME_BOARD_WIDTH));
+  }
+  if (
+    direction === "horizontal" &&
+    position[1] + SHIPS_LIST[shipIndex].size > GAME_BOARD_WIDTH
+  ) {
+    position[1] = Math.abs(position[1] - SHIPS_LIST[shipIndex].size);
+  } else if (position[0] + SHIPS_LIST[shipIndex].size > GAME_BOARD_WIDTH) {
+    position[0] = Math.abs(position[0] - SHIPS_LIST[shipIndex].size);
+  }
+};
+
+const addShip = () => {
+  createGameBoardArray();
+  for (let shipIndex = 0; shipIndex < SHIPS_LIST.length; shipIndex++) {
+    directionRoll();
```

Comment:
And the change here to `const direction = rollDirection();`. Later you just need to pass it to `makePosition(shipIndex, direction`. If I'm correct, you can event move it to this function and pass it to the `positionShipHorizontally` and `positionShipVertically`.

Reply:
If I'm right you cant put under the `const direction = rollDirection();` a `const position = createPosition(direction);` and then pass it to the `positionShipHorizontally ` and `positionShipVertically`